### PR TITLE
Feature/#cwb3x5 finite state machine

### PR DESCRIPTION
**Pull request by vvvictor07**
Simple state machine to handle behavior
- [x] Collision avoidance -  Avoid obstacles (Note: tagged on a Physics layer)
- [x] Flocking - Groups of AI move as a group simulating sheep herd behaviour 
- [x] Evade - Within their groups make them split and regroup avoiding danger 
- [x] Hide - AI hides behind objects as enemies pass by (Note: Physics layer would suit this best)
- [x] Wander - Chooses a random path to follow, explore a random area

### CODE REVIEW

File: Assets/Scripts/Scene 1/Behavior Scripts/StayInRadius.cs [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
+    public override Vector2 CalculateMoveSpeed(FlockAgent agent, List<Transform> context)
     {
-        //direction to the center
-        //magnitude will = distance
-        Vector2 centerOffet = center - (Vector2)agent.transform.position;
-        float t = centerOffet.magnitude / radius;
+        // direction to the center
+        // magnitude will = distance
+        var centerOffet = center - (Vector2)agent.transform.position;
+        var centerOffetInRadius = centerOffet.magnitude / radius;
```
Comment (nikitamedvedev) [inline Assets/Scripts/Scene 1/Behavior Scripts/StayInRadius.cs]: Offet -> Offset

File: Assets/Scripts/Scene 1/Behavior Scripts/SteeredCohesionBehavior.cs [hunk 2]
```diff
@@ -1,46 +1,58 @@
...
+
+    public float AgentSmoothTime = 0.5f;
+
+    private Vector2 currentVelocity;
 
-    public override Vector2 CalculateMove(FlockAgent agent, List<Transform> context, Flock flock)
+    public override Vector2 CalculateMoveSpeed(FlockAgent agent, List<Transform> context)
     {
-        //if no neighbours, return no adjustment
+        Debug.Log("SteeredCohesionBehavior " + GetInstanceID());
```
Comment (nikitamedvedev) [inline Assets/Scripts/Scene 1/Behavior Scripts/SteeredCohesionBehavior.cs]: Логи нужно оставить?

File: Assets/Scripts/Scene 1/Behavior Scripts/StayInRadius.cs (hunk 1 above)
Comment (vvvictor07) [reply Assets/Scripts/Scene 1/Behavior Scripts/StayInRadius.cs]: corrected

File: Assets/Scripts/Scene 1/Behavior Scripts/SteeredCohesionBehavior.cs (hunk 2 above)
Comment (vvvictor07) [reply Assets/Scripts/Scene 1/Behavior Scripts/SteeredCohesionBehavior.cs]: Исправлено

**Pull request merged**